crates/uv-resolver/src/requires_python.rs

```diff
@@ -169,6 +170,35 @@ impl RequiresPython {
     pub fn bound(&self) -> &Bound<Version> {
         &self.bound
     }
+
+    /// Returns this `Requires-Python` specifier as an equivalent marker
+    /// expression utilizing the `python_version` marker field.
+    ///
+    /// This is useful for comparing a `Requires-Python` specifier with
+    /// arbitrary marker expressions. For example, one can ask whether the
+    /// returned marker expression is disjoint with another marker expression.
+    /// If it is, then one can conclude that the `Requires-Python` specifier
+    /// excludes the dependency with that other marker expression.
+    ///
+    /// If this `Requires-Python` specifier has no constraints, then this
+    /// returns a marker tree that evaluates to `true` for all possible marker
+    /// environments.
+    pub fn to_marker_tree(&self) -> MarkerTree {
+        let mut conjuncts = vec![];
+        for spec in &*self.specifiers {
+            let expr_python_version = MarkerExpression::Version {
+                key: MarkerValueVersion::PythonVersion,
+                specifier: spec.clone(),
+            };
+            let expr_python_full_version = MarkerExpression::Version {
+                key: MarkerValueVersion::PythonFullVersion,
+                specifier: spec.clone(),
+            };
```

Comment:
I'm having trouble reasoning about whether we should be using both of these.

Reply:
Trying to talk it out... Let's say you have `Requires-Python = ">= 3.10.1"`...

Then we'd add `python_version >= "3.10.1"` and `python_full_version >= "3.10.1"`.

Then let's say we see a marker on a dependency `python_version == "3.10"`.

That actually _does_ intersect with `Requires-Python = ">= 3.10.1"`, because by definition in `python_version` we only capture the major and minor (it's computed as: `'.'.join(platform.python_version_tuple()[:2])`). But here we would consider it disjoint from `python_version >= "3.10.1"`.

Reply:
Ok, to simplify things, I might suggest this: use `self.bound` rather than `self.specifiers`. And add `>= self.bound`, and only include the major and minor release for the `MarkerValueVersion::PythonVersion` segment.

Reply:
The way I see it is if you have `foo ; python_full_version == '3.9'` and `Requires-Python: >=3.10`, then `foo` should be ignored.

That work by transforming `Requires-Python: >=3.10` to `python_version >= '3.10' and python_full_version >= '3.10'`. That marker expression in turn has zero overlap with `python_full_version == '3.9'`, so it is excluded. But if we instead had `bar ; python_full_version == '3.11'`, then there is possible overlap and thus they aren't considered disjoint.

Reply:
(Oops, when I wrote the above comment, I didn't see your two most recent comments. The GitHub UI is maybe slow to update.)

Reply:
Yeah, the basic problem is that `python_version` is definitionally truncated, and I worry that could lead to incorrect reasoning here.

Reply:
Yeah I'm going to use your example above to write a regression test.

Reply:
All righty, this should be fixed now! I added a regression test here: https://github.com/astral-sh/packse/pull/193